woocommerce/woocommerce-ios
# [Blaze] List and add payment methods

Closes: #12303


## Description

Blaze campaign creation flow

- Lists payment methods.
- Allows adding payment methods from the web view.

Changes
- Make changes to accommodate the API changes. We no longer receive add payment URLs from Blaze endpoint.
- Use the add payment form URL from the shipping labels flow for adding a payment method. 
- Load payment methods from the payment method list screen and add error handling. 
- Auto-select the newly added payment method by comparing the existing payments list. This is because we will no longer receive the new payment ID from the web view URL as a GET param value.  

## Testing instructions

List payment methods + Add a payment method
- Add a payment method (credit card) to your WPCOM account
- Login into a Woo store eligible for Blaze.
- Navigate to the Products tab and tap on a published product with no Blaze campaigns.
- You should see the "Promote with Blaze" button.
- Tap "Promote with Blaze" -> "Confirm Details". You should land on the payment screen.
- You should see the previously added credit card. (Added from the browser)
- Tap on the card and you will be navigated to the "Payment Method" screen.
- Tap on "Add another credit card" and follow the steps to add a new card.
- The web view should be dismissed automatically after adding the card. 
- You should see the newly added credit card.
- Check the WPCOM account from the browser and validate that the new payment has been added. 

Error alert
- [x] @selanthiraiyan Test the load payment methods error alert case using Proxyman to mock the error response.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/524475/c8a5f924-57a0-4cd1-b0bf-42a208a92a67


https://github.com/woocommerce/woocommerce-ios/assets/524475/0a0266f4-a27b-4679-9f4b-7ef224f92f39





---
